app/src/main/java/ch/epfl/qedit/backend/database/FirebaseDBService.java

```diff
@@ -1,49 +1,180 @@
+                                }
+                                responseCallback.onReceive(response);
+                            }
+                        });
+    }
+
+    @Override
+    public void getQuizTitle(String quizID, final Callback<Response<String>> responseCallback) {
+
+        final String language = Locale.getDefault().getLanguage();
+
+        db.collection("quizzes")
+                .document(quizID)
                 .get()
                 .addOnCompleteListener(
                         new OnCompleteListener<DocumentSnapshot>() {
                             @Override
                             public void onComplete(@NonNull Task<DocumentSnapshot> task) {
-                                Response<BundledData> response;
+                                Response<String> response;
                                 if (task.isSuccessful()) {
                                     DocumentSnapshot document = task.getResult();
-                                    if (Objects.requireNonNull(document).exists())
-                                        if (collection.equals("quizzes"))
-                                            response =
-                                                    Response.ok(
-                                                            new BundledData(document.getData()));
-                                        else response = Response.error(WRONG_COLLECTION);
-                                    else response = Response.error(WRONG_DOCUMENT);
-                                } else response = Response.error(CONNECTION_ERROR);
+                                    if (document.exists()) {
+                                        String title =
+                                                document.get("title_" + language, String.class);
+                                        response = Response.ok(title);
+                                    } else {
+                                        /**
+                                         * If the document does not exist then quizID does not
+                                         * describe an existing quiz in the database
+                                         */
+                                        response = Response.error(WRONG_DOCUMENT);
+                                    }
+                                } else {
+                                    /**
+                                     * If task is not successful, we don't retrieve any information
+                                     * with this query from the database
+                                     */
+                                    response = Response.error(CONNECTION_ERROR);
+                                }
                                 responseCallback.onReceive(response);
                             }
                         });
     }
+
+    public void getQuiz(final String quizID, final Callback<Response<Quiz>> responseCallback) {
```

Comment:
This method is correct, but I see one potential issue with it. We are sending the two requests one after the other, where we could in fact send both of them in parallel, which slows everything by a factor of two. That's why it is maybe cleaner to access the quiz in one go from the DB.

Reply:
I totaly agree with that, i thought about it but i did'nt succeed of making this parellel and clean, sorry. Maybe we could fixe that in a PR about optimization?

Reply:
Yes, that's fine to do it later ;)